xarray/coding/variables.py

```diff
@@ -145,11 +145,24 @@ class CFMaskCoder(VariableCoder):
     def encode(self, variable, name=None):
         dims, data, attrs, encoding = unpack_for_encoding(variable)
 
-        if encoding.get('_FillValue') is not None:
+        fv = encoding.get('_FillValue')
+        mv = encoding.get('missing_value')
+
+        if (fv is not None) and (mv is not None) and (fv != mv):
```

Comment:
Use `utils.equivalent()` for a NaN friendly equality check. Otherwise this would raise if you have `fv = mv = np.nan`.

Also, you don't need the extra parentheses -- `and` binds less tightly than most other operators.

Reply:
Done. Thanks.